tests/test_harness.rs

```diff
@@ -0,0 +1,63 @@
+use std::{
+    env::{self, join_paths},
+    fs::canonicalize,
+    io::{Error, ErrorKind, Result},
+    path::PathBuf,
+    process::Command,
+};
+
+pub struct Harness;
+
+impl Harness {
+    /// Build the project present in the current directory using `rustc` or `gccrs`
+    fn cargo_build(use_gccrs: bool) -> Result<()> {
+        let mut cmd = Command::new("cargo");
+
+        if use_gccrs {
+            // Add `gccrs` argument so that `cargo build` becomes `cargo gccrs build`
+            cmd.arg("gccrs");
+
+            // Tweak the path so that the most recently compiled *debug* version of
+            // cargo-gccrs is available as a subcommand
+            let mut paths = vec![canonicalize("../../target/debug")?];
```

Comment:
Using strings as paths with `/` in them will most likely lead to failures on windows. In Clippy we do this like this:

https://github.com/rust-lang/rust-clippy/blob/8d427b624f15a5c782c97021c3717451d86b8576/tests/cargo/mod.rs#L5-L8

Reply:
Good catch, thanks. I'm taking a look at clippy's implementation. Running tests on Windows is definitely a pain

Reply:
Fixed in 26174f9

Reply:
Did you try using `env::var_os("CARGO_TARGET_DIR")` to find the target dir?

Reply:
I've noticed that it always returns `Err(NotPresent)` in my tests, so I haven't looked much into it. Maybe I am doing something wrong?

To test it I simply added a `dbg!(std::env::var("CARGO_TARGET_DIR")); assert!(false)` before doing PATH manipulation in order to stop the test and display `stderr`.

Reply:
(Or `None` with `std::env::var_os("CARGO_TARGET_DIR")`)

Reply:
Hm, that's weird :thinking: The current implementation also LGTM

Reply:
I think I'll keep it like this for now if it works? It's not too disgusting and doesn't pose any problems for now, but we might refactor it later if necessary